lib/src/presentation/bloc/movie_controller.dart

```diff
@@ -50,4 +52,24 @@ class MovieController extends IMovieController {
 
   @override
   Stream<MovieState> get moviesStream => _moviesController.stream;
+
+  Future<void> searchMovie(
```

Comment:
for searching purpose , debounce method will help to improve performance. https://codewithflutter.com/how-to-debounce-textfield-onchange-in-dart/